src/components/Search/Search.jsx

```diff
@@ -1,34 +1,57 @@
-import React, { useState } from "react";
+
+import React, { useState, useRef, useEffect } from "react";
 import "./Search.css";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
 
 function Search({ onSearch }) {
   const [searchValue, setSearchValue] = useState("");
+  const [prevSearchValue, setPrevSearchValue] = useState("");
+  const searchInput = useRef(null);
 
   const handleInputChange = (event) => {
     setSearchValue(event.target.value);
   };
 
   const handleSearch = () => {
-    onSearch(searchValue);
+    if ( searchValue !== prevSearchValue) {
+      onSearch(searchValue);
+      setPrevSearchValue(searchValue);
+    }
   };
+
   const handleSearchOnEnter = (e) => {
     if (e.keyCode === 13) {
       handleSearch();
     }
   };
+
+  const handleSearchButtonClick = () => {
+    handleSearch();
+    if (searchValue === "") {
+      setSearchValue("");
+      setPrevSearchValue("");
+      onSearch("");
+    }
```

Comment:
```suggestion
    
```
remove this lines, we are checking same thing on `handleSearch()` and on clicking the icon you are calling same function two times. Removing the above mentioned line will fix it.